lib/docker/imageservice.go

```diff
@@ -646,3 +747,84 @@ func (s *remoteStore) updateRepo(ctx context.Context, remote, local distribution
 	_, err = remoteManifests.Put(ctx, manifest, distribution.WithTag(tag))
 	return trace.Wrap(err)
 }
+
+// multiScopeTokenHandler is a wrapper over the registry client token handler to allow additional scopes
+// to be added to an existing client. This works by re-instantiating the wrapped token handler if there
+// are additional scopes (repositories) that need to be added to the oauth token.
+type multiScopeTokenHandler struct {
+	tokenHandler registryauth.AuthenticationHandler
+	scopes       []registryauth.RepositoryScope
+	transport    http.RoundTripper
+	credentials  credentials
+}
+
+func (th *multiScopeTokenHandler) Scheme() string {
```

Comment:
is this implementing some external interface - would be good to know why APIs are exported?

Reply:
yes, it's an internal interface within docker.